### DISCUSSION TITLE
How to Using React ( Nextjs )

### DISCUSSION THREAD

**Discussion by bogordesaincom** [k-yle/rtsp-relay#177] [Q&A]
I'm using nextjs and follow instruction this [https://github.com/k-yle/rtsp-relay/blob/main/examples/react.md ](https://github.com/k-yle/rtsp-relay/blob/main/examples/react.md )but this result

<img width="1230" alt="Jepretan Layar 2022-10-06 pukul 19 00 52" src="https://user-images.githubusercontent.com/51335197/194307386-28fa398f-ab4f-4a6d-8430-c817c2386fa3.png">

**Comment by bogordesaincom** [comment]
this my code 

```js
  const canvas = useRef(null);

  const RelayRtsp = () => {
    if (!canvas.current) throw new Error('Ref is null');

    loadPlayer({
      url: 'ws://localhost:2000/api/stream/2',
      canvas: canvas.current,
    });
  };

  useEffect(() => {
    RelayRtsp();
  }, []);

  //   console.log(data);

  return (
    <>
      <Layout>
        <div className="canvas" ref={canvas}></div>
      </Layout>
      <Script src="https://cdn.jsdelivr.net/npm/rtsp-relay@1.6.1/browser/index.js" />
    </>
  );
```

**Comment by k-yle** [comment]
the react example works fine for me using nextjs. Could you post your entire code, not just that snippet please?

**Reply by k-yle** [reply]
example repository: https://github.com/k-yle/nextjs-rtsp-repro

**Reply by bogordesaincom** [reply]
are you using react 17? i'm using react 18

**Reply by k-yle** [reply]
i've updated [the example repository](https://github.com/k-yle/nextjs-rtsp-repro) to use react 18, which also seems to work fine
